File: java/universalChat/src/main/java/com/github/fernthedev/universal/encryption/PacketWrapper.java
package com.github.fernthedev.universal.encryption;

import lombok.Getter;

public class PacketWrapper<T> {



    protected boolean ENCRYPT = false;

    @Getter
    private T jsonObject;

    @Getter
    private String aClass;


    @Override
    public String toString() {
        return "PacketWrapper{" +
                ", aClass='" + aClass + '\'' +
                '}';
    }

    public PacketWrapper(T object, Class<?> aClass) {
        this.aClass = aClass.getName();
        this.jsonObject = object;
    }

//    @Deprecated
//    public PacketWrapper(@NonNull Object object) {
//        this.jsonObject = gson.toJson(object);
//        this.aClass = object.getClass().getName();
//    }

    public boolean encrypt() {
        return ENCRYPT;
    }



    protected PacketWrapper() {}
}

File: flutter/lib/backend/packetwrapper.dart
import 'dart:core';

import 'package:lombok/lombok.dart';

class PacketWrapper<T> {

  bool _ENCRYPT = false;

  @getter
  T _jsonObject;

  String _aClass;

  @override
  String toString() {
    return "PacketWrapper{" +
        ", aClass='" + _aClass + '\'' +
        '}';
  }

  PacketWrapper(T object, Type aClass) {
  this._aClass = aClass.runtimeType;
  this._jsonObject = object;
  }

//    @Deprecated
//    public PacketWrapper(@NonNull Object object) {
//        this.jsonObject = gson.toJson(object);
//        this.aClass = object.getClass().getName();
//    }

  bool encrypt() {
    return _ENCRYPT;
  }



  _PacketWrapper() {}
}

class something<T> {
  bool _ENCRYPT = false;

  @getter
  T _jsonObject;

  String _aClass;


}